install/helm/agones/templates/extensions.yaml

```diff
@@ -11,11 +11,22 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-{{- $ca := genCA "admission-controller-ca" 3650 }}
-{{- $cn := printf "agones-controller-service" }}
-{{- $altName1 := printf "agones-controller-service.%s"  .Release.Namespace }}
-{{- $altName2 := printf "agones-controller-service.%s.svc" .Release.Namespace }}
-{{- $cert := genSignedCert $cn nil (list $altName1 $altName2) 3650 $ca }}
+{{- $cert := "" }}
+{{- $key := "" }}
+{{- if .Values.agones.controller.generateTLS }}
+  {{- $previous := lookup "v1" "Secret" .Release.Namespace (include "agones.fullname" . | printf "%s-cert") }}
+  {{- if $previous | and .Values.agones.controller.reuseGeneratedTLS }}
+    {{- $cert = index $previous.data "server.crt" | b64dec }}
+    {{- $key = index $previous.data "server.key" | b64dec }}
+  {{- else }}
+    {{- $cn := printf "agones-controller-service.%s.svc" .Release.Namespace }}

```

Comment:
Shouldn't this still be `agones-controller-service` (as it was before)? Right now the common name is the same as the altName2.

Reply:
https://kubernetes.io/docs/tasks/extend-kubernetes/setup-extension-api-server/

`5. Create a server cert/key for the api-server to use for HTTPS. This cert should be signed by the above CA. It should also have a CN of the Kube DNS name. This is derived from the Kubernetes service and be of the form <service name>.<service name namespace>.svc`

And it is also said inside file generating certs `install/helm/agones/certs/cert.sh` - `echo "Common Name should be: agones-controller-service.agones-system.svc"`